design/use-ironic.md

```diff
@@ -0,0 +1,151 @@
+         * [Version Skew Strategy](#version-skew-strategy)
+      * [Drawbacks [optional]](#drawbacks-optional)
+      * [Alternatives [optional]](#alternatives-optional)
+      * [References](#references)
+
+<!-- Added by: stack, at: 2019-02-18T16:27-05:00 -->
+
+<!--te-->
+
+## Summary
+
+Part of being a bare metal provisioning system is having the ability
+to control and provision images to bare metal hosts. Many such systems
+exist, so rather than build a new one we want to select an existing on
+that supports our needs. Initially this will be
+[Ironic](https://docs.openstack.org/ironic/latest/).
+
+## Motivation
+
+We want to choose a tool that has an API to drive, provides good
+community support, can handle a variety of hardware, and can run in a
+Kubernetes environment. Ironic meets all of those criteria.
+
+### Goals
+
+1. Choose an existing provisioning/deployment tool to be used to
+   configure hosts and their images.
+2. Create an architecture that allows us to choose a different tool
+   later, if we have some reason to do that.
+
+### Non-Goals
+
+1. Build a new provisioning/deployment tool for the sake of having one
+   written in golang.
+2. Integrate with every provisioning/deployment tool potential users
+   of MetalKube might have in their data centers.
+
+## Proposal
+
+Package Ironic as an image in a way that allows us to drive it from an
+operator, but hide its use from the end user.
+
+### User Stories [optional]
+
+None
+
+### Implementation Details/Notes/Constraints [optional]
+
+- We want to eliminate the need for a MySQL database or RabbitMQ
+  message bus.
+- We need to include Ironic Inspector as well as the main Ironic API
+  so we can perform hardware introspection.
+- Both services should run inside the same Pod as the
+  baremetal-operator.
+- The Pod containing the services will need host network access to be
+  able to run a PXE service on the provisioning network.
+
+### Risks and Mitigations
+
+Giving the Pod host network access gives it the ability to see much
+more of the host on which it is running. On the other hand, this is a
+provisioning tool, and it's going to be writing images to hosts. So it
+already has quite a lot of power.
+
+## Design Details
+
+- The YAML file with the specification for the Pod to run the operator
+  needs to be expanded to include the other container(s) needed for
+  Ironic.
+
+### Work Items
+
+TBD
+
+### Dependencies
+
+- Ironic image
+- Ironic Inspector image
+
+### Test Plan
+
+We will eventually run end-to-end tests using Ironic and the operator.
+
+Ironic itself will be tested using the standard OpenStack testing
+resources.
+
+### Upgrade / Downgrade Strategy
+
+TBD
+
+### Version Skew Strategy
+
+Version skew should not apply if the operator and ironic services run
+in the same Pod.
+
+## Drawbacks [optional]
+
+Ideally we would not need a long running service at all, but until we
+have a tool compatible with the Job API we cannot avoid that.
```

Comment:
Maybe this is where we could expand on how even though Ironic is a long running service, it's not the primary owner of any given information.  If it was lost, we can reconfigure it from state stored in CRDs.

Reply:
Done

Reply:
Are you referring to the expanded text about MySQL in the details section earlier in the doc?

Reply:
Yeah, I thought having it up in the implementation details section made more sense, sorry I should have been explicit in my "done" message. Do you think I should add more here, too?

Reply:
All good.  thanks!